vic/drivers/cesm/bld/build_vic_namelist

```diff
@@ -0,0 +1,175 @@
+
+    # copy VIC files to confdir
+    for key in ['vic_global_param', 'vic_constants']:
+        grid_config[key] = copy_to_vicconf(coderoot, confdir, grid_config[key])
+
+    # copy Constants and Global Parameters to RUNDIR
+    copy_to_rundir(grid_config, caseid, rundir, casedocs)
+
+    # write input files list
+    dst_file = os.path.join(casebld, 'vic.input_data_list')
+
+    # if file exists so add temporarily write permissions
+    if os.path.isfile(dst_file):
+        os.chmod(dst_file, stat.S_IWUSR)
+
+    write_input_data_list(grid_config, dst_file)
+    set_readonly(dst_file)
+
+
+def copy_to_rundir(grid_config, caseid, rundir, casedocs):
+    for filekey, header_txt in [('vic_constants', 'Constants'),
+                                ('vic_global_param', 'Global Parameters')]:
+        header = header_template.format(header_txt, caseid)
+        fname = os.path.basename(grid_config[filekey])
+        dst_file = os.path.join(rundir, fname)
+
+        # if file exists so add temporarily write permissions
+        if os.path.isfile(dst_file):
+            os.chmod(dst_file, stat.S_IWUSR)
+
+        # copy file
+        copy_clean_vic_config(grid_config[filekey], dst_file,
+                              header=header, rundir=rundir, **grid_config)
+
+        # update the grid config
+        grid_config[filekey] = dst_file
+
+        # change permissions to read only
+        set_readonly(dst_file)
+
+        if casedocs:
+            doc_file = os.path.join(casedocs, fname)
+            if os.path.isfile(doc_file):
+                os.remove(doc_file)
+            shutil.copy(dst_file, doc_file)
+            set_readonly(doc_file)
+
+
+def set_readonly(fname):
+    os.chmod(fname, stat.S_IRUSR | stat.S_IRGRP | stat.S_IROTH)
+
+
+def read_config(config_file):
+    """
+    Return a dictionary with subdictionaries of all configFile options/values
+    """
+
+    if not os.path.isfile(config_file):
+        raise VIC_Error('Config File {0} does not exist'.format(config_file))
+
+    config = SafeConfigParser()
+    config.optionxform = str
+    config.read(config_file)
+    sections = config.sections()
+    dict1 = {}
+    for section in sections:
+        options = config.options(section)
+        dict2 = {}
+        for option in options:
+            dict2[option] = config.get(section, option)
+        dict1[section] = dict2
+
+    return dict1
+
+
+def copy_clean_vic_config(src, dst, header=None, **kwargs):
+    """Copy VIC style ASCII configuratoin file from src to dst
+
+        remove comments and empty lines
+    """
+    with open(src, 'r') as fsrc:
+        with open(dst, 'w') as fdst:
+            lines = fsrc.readlines()
+            if header is not None:
+                fdst.write(header)
+            for line in lines:
+                line = line.format(**kwargs)
+                line = os.path.expandvars(line.split('#', 1)[0].strip())
+                if line:
+                    fdst.write(line + '\n')
+
```

Comment:
using the `with open(...` syntax allows the implicit closing of the file.  This is called a context manager and it will automatically close the files (`fsrc` and `fdst`) when we leave their respective `with ...` blocks.